File: launch_school/RB101-Foundations/small_problems/multiply_all_pairs.rb
def multiply_all_pairs(arr1, arr2)
  final_arr = []
  arr1.each do |item|
    arr2.each do |element|
      final_arr << item * element
    end
  end
  final_arr.sort
end

p multiply_all_pairs([2, 4], [4, 3, 1, 2]) == [2, 4, 4, 6, 8, 8, 12, 16]

File: launch_school/LS215-Computational-Thinking/small_problems/list_processing/multiply_all_pairs.js
function multiplyAllPairs(arr1, arr2) {
  let collection = [];

  arr1.forEach(firstNum => {
    arr2.forEach(secondNum => {
      let product = firstNum * secondNum;
      collection.push(product);
    });
  });

  return collection.sort((a, b) => a - b);
}

let result = multiplyAllPairs([2, 4], [4, 3, 1, 2]);    // [2, 4, 4, 6, 8, 8, 12, 16]

console.log(result);
